src/imu_control_parameters.cpp

```diff
@@ -0,0 +1,333 @@
+    }
+    return hasError;
+}
+
+bool ImuControlParameters::triggerCommand(std::string& command_to_execute, bool hasError)
+{
+    if(command_to_execute == "software_reset")
+    {
+        int ret = m_iioWrapper.set_software_reset();
+
+        if(ret == 0)
+        {
+             RCLCPP_INFO(rclcpp::get_logger("rclcpp_set_parameter"), "command_to_execute: software_reset");
+
+             // reset our member values also
+             m_operation_mode = 0;
+             m_filter_size = 0;
+             m_burst_size_selection = 0;
+             m_burst_data_selection = 0;
+             m_linear_acceleration_compensation = 1;
+             m_point_of_percussion_alignment = 1;
+             m_internal_sensor_bandwidth = 0;
+             m_sync_mode_select = 3;
+             m_sync_polarity = 0;
+             m_data_ready_polarity = 1;
+             m_sync_signal_scale = 2000;
+             m_decimation_filter = 0;
+             m_accel_calibbias_x = 0;
+             m_accel_calibbias_y = 0;
+             m_accel_calibbias_z = 0;
+             m_anglvel_calibbias_x = 0;
+             m_anglvel_calibbias_y = 0;
+             m_anglvel_calibbias_z = 0;
+             m_sampling_frequency = 2000;
+
+             m_command_to_execute = "no_command";
+
+             // reset values also from node
+             m_node->set_parameter(rclcpp::Parameter("operation_mode", 0));
+             m_node->set_parameter(rclcpp::Parameter("filter_size", 0));
+             m_node->set_parameter(rclcpp::Parameter("burst_size_selection", 0));
+             m_node->set_parameter(rclcpp::Parameter("burst_data_selection", 0));
+             m_node->set_parameter(rclcpp::Parameter("linear_acceleration_compensation", 1));
+             m_node->set_parameter(rclcpp::Parameter("point_of_percussion_alignment", 1));
+             m_node->set_parameter(rclcpp::Parameter("internal_sensor_bandwidth", 0));
+             m_node->set_parameter(rclcpp::Parameter("sync_mode_select", 3));
+             m_node->set_parameter(rclcpp::Parameter("sync_polarity", 0));
+             m_node->set_parameter(rclcpp::Parameter("data_ready_polarity", 1));
+             m_node->set_parameter(rclcpp::Parameter("sync_signal_scale", 2000));
+             m_node->set_parameter(rclcpp::Parameter("decimation_filter", 0));
+             m_node->set_parameter(rclcpp::Parameter("accel_calibbias_x", 0));
+             m_node->set_parameter(rclcpp::Parameter("accel_calibbias_y", 0));
+             m_node->set_parameter(rclcpp::Parameter("accel_calibbias_z", 0));
+             m_node->set_parameter(rclcpp::Parameter("anglvel_calibbias_x", 0));
+             m_node->set_parameter(rclcpp::Parameter("anglvel_calibbias_y", 0));
+             m_node->set_parameter(rclcpp::Parameter("anglvel_calibbias_z", 0));
+             m_node->set_parameter(rclcpp::Parameter("sampling_frequency", 2000));
+             m_node->set_parameter(rclcpp::Parameter("command_to_execute", "no_command"));
+        }
+        else
```

Comment:
this is not the handling I would expect. the software reset will indeed perform a reset of the device but you should not manually set the parameters when this occurs, instead these parameters should be updated from the hardware itself